streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java

```diff
@@ -245,9 +244,10 @@ private void updateTaskCommitMetadata(final Map<TopicPartition, OffsetAndMetadat
     }
 
     private void commitSuccessfullyProcessedTasks() {
-        if (processingMode == AT_LEAST_ONCE && !tasks.successfullyProcessed().isEmpty()) {
-            log.error("Streams encountered an error when processing tasks." +
-                " Will commit all previously successfully processed tasks");
+        if (!tasks.successfullyProcessed().isEmpty()) {
+            log.info("Streams encountered an error when processing tasks." +
+                " Will commit all previously successfully processed tasks {}",
+                tasks.successfullyProcessed().toString());
```

Comment:
For future reference, you can skip the `toString()` -- the logger should make the conversion implicitly